tests/javascript/frontediting/spec.js

```diff
@@ -0,0 +1,208 @@
+		});
+
+		it('Should detach all selected elements from jEditMakeAbsolute div', function () {
+			expect($('#jEditMakeAbsolute')).not.toContainElement('.fr-ed-make-abs-true');
+		});
+
+		it('Should append all detached elements to body', function () {
+			expect($('body>div.fr-ed-make-abs-true').length).toEqual(2);
+		});
+	});
+
+	describe('Frontediting jEditMakeAbsolute with argument false', function () {
+		beforeAll(function () {
+			$('.fr-ed-make-abs-false').jEditMakeAbsolute(false);
+		});
+
+		it('Should change style of all selected elements ', function () {
+			var $elements = $('.fr-ed-make-abs-false');
+			var $first = $elements.first();
+			var $second = $elements.last();
+
+			expect($first).toHaveCss({position: "absolute", marginLeft: '0px', marginTop: '0px'});
+			expect($second).toHaveCss({position: "absolute", marginLeft: '0px', marginTop: '0px'});
+		});
+
+		it('Should not detach any selected elements from jEditMakeAbsolute div', function () {
+			expect($('#jEditMakeAbsolute')).toContainElement('.fr-ed-make-abs-false');
+		});
+	});
+
+	describe('Frontediting modules jmoddiv on mouseenter', function () {
+		beforeAll(function () {
+			$('#jmoddiv-module').mouseenter();
+			this.$editBtn = $('body>a.btn.jmodedit');
+		});
+
+		it('Should add class jmodinside to the div', function () {
+			expect($('#jmoddiv-module')).toHaveClass('jmodinside');
+		});
+
+		it('Should prepend edit icon element and move it to end of body', function () {
+			expect(this.$editBtn).toExist();
+		});
+
+		it('Should set jmodedit class on the added element', function () {
+			expect(this.$editBtn).toHaveClass('jmodedit');
+		});
+
+		it('Should set attribute href = "url" on the added element', function () {
+			expect(this.$editBtn).toHaveAttr('href', 'url');
+		});
+
+		it('Should set attribute title = "title" on the added element', function () {
+			expect(this.$editBtn).toHaveAttr('data-original-title', 'mod-title');
+		});
+
+		it('Should set attribute target = "target" on the added element', function () {
+			expect(this.$editBtn).toHaveAttr('target', 'target');
+		});
+
+		describe('Edit icon on mouseenter', function () {
+			beforeAll(function () {
+				this.$editBtn.mouseenter();
+			});
+
+			it('Should show tooltip', function () {
+				expect($('body>div.tooltip')).toExist();
+			});
+			it('Should set content as "title" inside tooltip', function () {
+				expect($('.tooltip-inner').first()).toHaveText('mod-title');
+			});
+		});
+
+		describe('Edit icon on mouseleave', function () {
+			beforeEach(function () {
+				jasmine.clock().install();
+				this.$editBtn.mouseleave();
+			});
+
+			afterEach(function () {
+				jasmine.clock().uninstall();
+			});
+
+			it('Should remove tooltip and button after a 500 millisecond delay', function () {
+				expect($('body>div.tooltip')).toExist();
+				expect($('body>a.btn.jmodedit')).toExist();
+
+				jasmine.clock().tick(501);
```

Comment:
I feel its good  to declare a constant with usage of the tick so the code is easily readable